secure_message/resources/threads.py

```diff
@@ -20,13 +23,68 @@ def get(thread_id):
         logger.info("Getting messages from thread", thread_id=thread_id, user_uuid=g.user.user_uuid)
 
         conversation = Retriever().retrieve_thread(thread_id, g.user)
+        conversation_metadata = Retriever().retrieve_thread_metadata(thread_id)
 
         logger.info("Successfully retrieved messages from thread", thread_id=thread_id, user_uuid=g.user.user_uuid)
         messages = []
         for message in conversation.all():
             msg = message.serialize(g.user, body_summary=False)
             messages.append(msg)
-        return jsonify({"messages": add_users_and_business_details(messages)})
+
+        if conversation_metadata.is_closed:
+            return jsonify({"messages": add_users_and_business_details(messages),
+                            "metadata": ThreadById._build_metadata(conversation_metadata)})
+        else:
+            return jsonify({"messages": add_users_and_business_details(messages)})
+
+    @staticmethod
+    def _build_metadata(conversation_metadata):
+        return {
+            "is_closed": conversation_metadata.is_closed,
+            "closed_by": conversation_metadata.closed_by,
+            "closed_by_uuid": conversation_metadata.closed_by_uuid,
+            "closed_on": conversation_metadata.closed_on
+        }
+
+    @staticmethod
+    def patch(thread_id):
+        """Modify conversation metadata"""
+
+        logger.info("Attempting to modify metadata for thread", thread_id=thread_id, user_uuid=g.user.user_uuid)
+        request_data = request.form
+        msg_property, value = ThreadById._validate_request(request_data)
+
+        logger.info("Getting metadata for thread", thread_id=thread_id, user_uuid=g.user.user_uuid)
+        metadata = Retriever.retrieve_thread_metadata(thread_id)
+        if msg_property == 'is_closed':
+            if value:
+                if metadata.is_closed:
+                    logger.info("Conversation already closed", thread_id=thread_id, user_uuid=g.user.user_uuid)
```

Comment:
Do we even care what the request data is if the conversation is closed? If not we could make this the first thing we check.